plutus-core/plutus-ir/test/TransformSpec.hs

```diff
@@ -246,6 +264,30 @@ inline =
             , "letNonPureMultiStrict"
             ]
 
+-- | Check whether a term is globally unique.
+checkUniques :: (Ord a, MonadError (UniqueError a) m) => Term TyName Name uni fun a -> m ()
+checkUniques =
+    Uniques.checkTerm (\case { MultiplyDefined{} -> True; _ -> False})
+
+-- | Tests that the inliner doesn't incorrectly capture variable names.
+nameCapture :: TestNested
+nameCapture =
+    let goldenInlineUnique :: Term TyName Name PLC.DefaultUni PLC.DefaultFun PLC.SrcSpan ->
+            IO String
+        goldenInlineUnique pir =
+            rethrow . asIfThrown @(UniqueError PLC.SrcSpan) $ do
+                let pirInlined = runQuote $ do
+                        renamed <- PLC.rename pir
+                        Inline.inline mempty def renamed
+                -- Make sure the inlined term is globally unique.
+                _ <- checkUniques pirInlined
+                pure . render $ prettyPirReadable pirInlined
+    in
+    testNested "nameCapture" $
+        map
+            (goldenPirMUnique goldenInlineUnique pTerm)
+            [ "nameCapture"]
```

Comment:
Why not just add this to the other list? You've duplicated the whole `goldenInlineUnique` bit just to put this in another test tree? You could just add `++ [ goldenPirMUnique goldenInlineUnique pterm "nameCapture" ]` to the end of the thing above, no?

Reply:
it doesn't work because if I do that it won't show the uniques. The two `goldenInlineUnique` are actually not the same, the one for the name capture print things out different. I'll change the name.